lib/logthrsource/logthrsourcedrv.c

```diff
@@ -351,30 +378,30 @@ log_threaded_source_close_batch(LogThreadedSourceDriver *self)
 }
 
 void
-log_threaded_source_post(LogThreadedSourceDriver *self, LogMessage *msg)
+log_threaded_source_worker_post(LogThreadedSourceWorker *self, LogMessage *msg)
 {
   msg_debug("Incoming log message",
             evt_tag_str("input", log_msg_get_value(msg, LM_V_MESSAGE, NULL)),
+            evt_tag_str("driver", self->control->super.super.id),
+            evt_tag_int("worker_index", self->worker_index),
             evt_tag_msg_reference(msg));
-  _apply_message_attributes(self, msg);
-  log_source_post(&self->worker->super, msg);
+  _apply_message_attributes(self->control, msg);
+  log_source_post(&self->super, msg);
 
-  if (self->auto_close_batches)
-    log_threaded_source_close_batch(self);
+  if (self->control->auto_close_batches)
+    log_threaded_source_close_batch(self->control);
```

Comment:
the log_threaded_source_close_batch() function should also be converted to be a method of the Worker.

Reply:
I'd probably also move these set of functions to the rest of the Worker methods.

Reply:
Thanks, I have transformed the `close_batch()` function to operate on the worker.